contracts/tg4-stake/src/contract.rs

```diff
@@ -646,20 +737,52 @@ mod tests {
         instantiate(deps, mock_env(), info, msg).unwrap();
     }
 
-    fn bond(
-        mut deps: DepsMut<TgradeQuery>,
+    // Helper for staking only liquid assets
+    fn bond_liquid(
+        deps: DepsMut<TgradeQuery>,
+        user1: u128,
+        user2: u128,
+        user3: u128,
+        height_delta: u64,
+    ) {
+        bond(deps, (user1, 0), (user2, 0), (user3, 0), height_delta);
+    }
+
+    // Helper for staking only illiquid assets
+    fn bond_vesting(
+        deps: DepsMut<TgradeQuery>,
         user1: u128,
         user2: u128,
         user3: u128,
         height_delta: u64,
+    ) {
+        bond(deps, (0, user1), (0, user2), (0, user3), height_delta);
+    }
+
+    // Full stake is composed of `(liquid, illiquid (vesting))` amounts
+    fn bond(
+        mut deps: DepsMut<TgradeQuery>,
+        user1_stake: (u128, u128),
+        user2_stake: (u128, u128),
+        user3_stake: (u128, u128),
+        height_delta: u64,
     ) {
         let mut env = mock_env();
         env.block.height += height_delta;
 
-        for (addr, stake) in &[(USER1, user1), (USER2, user2), (USER3, user3)] {
-            if *stake != 0 {
-                let msg = ExecuteMsg::Bond {};
-                let info = mock_info(addr, &coins(*stake, DENOM));
+        for (addr, stake) in &[
+            (USER1, user1_stake),
+            (USER2, user2_stake),
+            (USER3, user3_stake),
+        ] {
+            if stake.0 != 0 || stake.1 != 0 {
+                let vesting_tokens = if stake.1 != 0 {
+                    Some(coin(stake.1, DENOM))
```

Comment:
Haha, the same check doubled bothers me, but at the same time I can't figure out equally simple solution.

Reply:
Being there.